ldms/src/core/ldms_xprt.c

```diff
@@ -347,34 +347,42 @@ static void send_dir_update(struct ldms_xprt *x,
 {
 	size_t hdr_len;
 	size_t buf_len;
-	size_t cnt;
+	size_t json_data_len;
 	struct ldms_reply *reply;
 
 	assert(t != LDMS_DIR_LIST);
 
 	hdr_len = sizeof(struct ldms_reply_hdr)
 		+ sizeof(struct ldms_dir_reply);
-	buf_len = ldms_xprt_msg_max(x);
-	reply = malloc(buf_len);
+	/* Calculate the size of the string */
+	json_data_len = snprintf(NULL, 0, "{ \"directory\" : [ %s ]}", json);
```

Comment:
Not the most efficient way to do this, but let's start with this.

Reply:
I have changed this to
`
json_data_len  = strlen("{ ....}") + json_sz;
`